src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
If I change the `Validation` definition to `public interface Validation<E,T> extends Applicative<T>`, then my apply looks like this `<U> Applicative<U> apply(Applicative<? extends Function<? super T, ? extends U>> f)`. Which is what I was looking for. But this isn't right because we're saying that the type `T` is the `Applicative`, which it is not. `Validation` is the `Applicative`.